src/shared/tpm2-util.c

```diff
@@ -1598,10 +1793,25 @@ int tpm2_seal(const char *device,
         *ret_pcr_bank = pcr_bank;
         *ret_primary_alg = primary_alg;
 
+        if (ret_srk_buf) {
+                /*
+                 * make a copy since we don't want the caller to understand that
+                 * ESYS allocated the pointer. It would make tracking what deallocator
+                 * to use for srk_buf in which context a PITA.
+                 */
+                void *tmp = memdup(srk_buf, srk_buf_size);
+                if (!tmp)
+                        return log_oom();
+
+                *ret_srk_buf = tmp;
+                *ret_srk_buf_size = srk_buf_size;
+        }
```

Comment:
like int the other cases, if the memdup() here fails you already touched ret_secret and all the other stuff. hence please move this block just a bit up, *before* the `*ret_secret = TAKE_PTR(…` line, and things are in order again